hash/hash_comparator.go

```diff
@@ -105,3 +134,38 @@ func decodeArgon2idHash(encodedHash string) (p *config.Argon2, salt, hash []byte
 
 	return p, salt, hash, nil
 }
+
+// decodePbkdf2Hash decodes PBKDF2 encoded password hash.
+// format: $pbkdf2_<algorithm>$c=<iteration>$<salt>$<hash>
+func decodePbkdf2Hash(encodedHash string) (p *Pbkdf2, salt, hash []byte, err error) {
```

Comment:
Where is this function body coming from and what standard does it rely on? :) Is there maybe some well-tested library we can use for this?

Reply:
I could not found any well-tested Go libraries 😢 

This encoding structure is coming from [Django](https://docs.djangoproject.com/en/3.2/topics/auth/passwords/) and [passlib](https://passlib.readthedocs.io/en/stable/lib/passlib.hash.pbkdf2_digest.html) and implementation is from [Django](https://github.com/django/django/blob/8036b53de61f16098a2f22c50621987182bdfaed/django/contrib/auth/hashers.py#L265-L317).
In former commit, iteration part ( `c=<iteration>` ) refer to argon2 in kratos but I changed it Django style at 75b8fcd.

Reply:
Hm, what about other languages and frameworks? Are all of them using this syntax?

Reply:
I found more PBKDF2 hasher in 2 framework but all are different syntax.

In case of Symfony(PHP), salt is stored another column.
https://github.com/symfony/symfony/blob/88f3ef62d6ab870128352c8686c7889562698faf/src/Symfony/Component/Security/Core/Encoder/UserPasswordEncoder.php#L37

In case of Spring Boot(Java), syntax is `<salt><hash>` without a separator.
https://github.com/spring-projects/spring-security/blob/main/crypto/src/main/java/org/springframework/security/crypto/password/Pbkdf2PasswordEncoder.java#L201

Reply:
Hm that would then be part of the documentation/endpoint for importing those hashes I guess. The django format makes most sense IMO, as it includes all parameters.

Reply:
Ok I see - so we would need to support multiple formats in the end to be able to import user passwords. I guess we can live with that!

Reply:
Does it mean "Store single format per algorithm, support multiple formats in the end"?

I wrote documentation about format at the commit 060cbdb.

Reply:
Yeah, I think we probably need to do some translation at some point, but for now the approach you took is fine :)